### PR TITLE
Add sex check

### PR DESCRIPTION
**Pull request by Lucpen**
## PR checklist

- [x] Added sex check from bam file ngsbits sample gender
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test_one_sample,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by Lucpen** [comment]
I am currently running with the full data set to see if it works, as the test one does not contain chrX and chrysalis

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - New parameters to skip fastqc and haplocheck (`--skip_fastqc` and `--skip_haplocheck`) [#438](https://github.com/nf-core/raredisease/pull/438)
 - CNVnator for copy number variant calling [#438](https://github.com/nf-core/raredisease/pull/434)
 - A new parameter `svdb_query_bedpedbs` to provide bedpe files as databases for SVDB query [#449](https://github.com/nf-core/raredisease/pull/449)
+- ngsbits samplegender to check sex
```
Comment (Lucpen) [inline CHANGELOG.md:19]:
```suggestion
- ngsbits samplegender to check sex [#453] (https://github.com/nf-core/raredisease/pull/453)
```

File: conf/modules/qc_bam.config [hunk 2]
```diff
@@ -53,6 +53,16 @@ process {
         ext.prefix = { "${meta.id}_mosdepth" }
     }
 
+    withName: '.*QC_BAM:NGSBITS_SAMPLEGENDER' {
+        //  NGSBITS_SAMPLEGENDER needs a chrX and chrY in order to run so we skip it for the two test profiles
+        ext.when = { !(workflow.profile.tokenize(',').intersect(['test', 'test_one_sample', 'test_sentieon']).size() >= 1) || workflow.stubRun }
```
Comment (jemten) [inline conf/modules/qc_bam.config:58]: Let's add a prefix to the output file for clarity. Could be something like:
```
ext.prefix = { "${meta.id}_sex }
```
or 
```
ext.prefix = { "${meta.id}_ngsbits_sex }
```

File: subworkflows/local/qc_bam.nf [hunk 3]
```diff
@@ -59,6 +60,9 @@ workflow QC_BAM {
         SENTIEON_WGSMETRICS ( ch_bam_bai, ch_genome_fasta, ch_genome_fai, ch_intervals_wgs.map{ interval -> [[:], interval]} )
         SENTIEON_WGSMETRICS_Y ( ch_bam_bai, ch_genome_fasta, ch_genome_fai, ch_intervals_y.map{ interval -> [[:], interval]} )
 
+        // Check sex
+        NGSBITS_SAMPLEGENDER(ch_bam_bai, ch_genome_fasta, ch_genome_fai, "xy")
```
Comment (jemten) [inline subworkflows/local/qc_bam.nf:64]: We could consider making the method a parameter, ie `"xy"` becomes something like "ngsbits_samplegender_method" (or something that rolls of the tongue more easily), with the default set to `xy`.

**Review by jemten — COMMENTED** [review]
Nice work! A couple of small comments and suggestions. Let me know what you think of parameterising the ngbits method, we could leave it as is for now.

File: conf/modules/qc_bam.config (hunk 2 above)
Comment (Lucpen) [reply conf/modules/qc_bam.config:58]: Done! :)

File: subworkflows/local/qc_bam.nf (hunk 3 above)
Comment (Lucpen) [reply subworkflows/local/qc_bam.nf]: Done

**Review by jemten — APPROVED** [review]
Looks good!

**Pull request merged**